adapters/repos/db/index.go

```diff
@@ -1484,3 +1518,22 @@ func objectSearchPreallocate(limit int, shards []string) ([]*storobj.Object, []f
 
 	return objects, scores
 }
+
+func (i *Index) addNewShard(ctx context.Context,
+	class *models.Class, shardName string,
+) error {
+	// TODO: locking???
```

Comment:
I think that Red's #3035 PR addresses this issue, so if we merge his changes first then you could rebase your PR and use Red's solution